src/main/java/org/broadinstitute/hellbender/tools/exome/ExomeReadCounts.java

```diff
@@ -236,6 +319,92 @@ public void onTraversalStart() {
     }
 
     /**
+     * Builds the exon collection given the values of user arguments.
+     *
+     * @throws UserException if there is some inconsistency in user arguments and inputs.
+     * @throws GATKException if there was any problem parsing the content of the exome file.
+     * @return never {@code null}.
+     */
+    private ExonCollection<? extends Locatable> resolveExonCollection() {
+        final ExonCollection<? extends Locatable> result;
+        if (exomeFile != null) {
+            result = resolveExonCollectionFromExomeFile();
+        } else if (hasIntervals()) {
+            final SAMSequenceDictionary sequenceDictionary = getBestAvailableSequenceDictionary();
+            if (sequenceDictionary == null) {
+                throw new UserException("missing reference sequence dictionary, as no input contains such a dictionary... please consider to provide fail-over dictionary using " +
+                        "a reference (e.g. --reference my_reference.fasta)");
+            }
+            result = ExonCollections.fromSimpleIntervalList(intervalArgumentCollection.getIntervals(sequenceDictionary));
+        } else {
+            throw new UserException(String.format("You must indicate the set of exon as input intervals (e.g. -L target-intervals.list) or a exome feature file (e.g. -%s my-targets.bed) ",EXOME_FILE_SHORT_NAME));
```

Comment:
With the intervals one saves time in the traversal but given a exome-file it isn't in fact needed.